File: fsharp/kindergarten-garden/KindergartenGarden.fs
module KindergartenGarden

type Plant = 
| Clover
| Grass
| Radishes
| Violets

let students = 
    [   "Alice", 0;
        "Bob", 1;
        "Charlie", 2;
        "David", 3;
        "Eve", 4;
        "Fred", 5;
        "Ginny", 6;
        "Harriet", 7;
        "Ileana", 8;
        "Joseph", 9;
        "Kincaid", 10;
        "Larry", 11]
    |> dict

let findPlant abbrev = 
    match abbrev with
    | 'C' -> Plant.Clover
    | 'G' -> Plant.Grass
    | 'R' -> Plant.Radishes
    | 'V' -> Plant.Violets

let plants (diagram: string) student = 
    let rows = diagram.Split '\n'
    let studentVal = 2 * students.[student]
    let planted = rows.[0].[studentVal..(studentVal+1)] + 
                  rows.[1].[studentVal..(studentVal+1)]
    planted |> Seq.toList |> List.map findPlant

File: r/kindergarten-garden/1/kindergarten-garden.R
library(tidyverse)

students <- c("Alice", "Bob", "Charlie", "David", "Eve", "Fred", "Ginny",
               "Harriet", "Ileana", "Joseph", "Kincaid", "Larry")

species <- function(code) {
  switch(code,
         G = "Grass",
         C = "Clover",
         R = "Radish",
         V = "Violet")
}

plants <- function(garden, child) {
  rows <- garden |> str_split_1("\\n") |> str_split("")
  start_col <- 2 * match(child, students) - 1
  owns <- c(rows[[1]][start_col:(start_col + 1)], 
            rows[[2]][start_col:(start_col + 1)])
  owns |> sapply(species) |> unname()
}
